sample-app/src/pages/UniversalSearchPage.tsx

```diff
@@ -17,7 +19,14 @@ export default function UniversalSearchPage(props: { universalResultsConfig: Uni
       vertical: {}
     })
     answersActions.setVerticalKey('');
-    answersActions.executeUniversalQuery();
+    const executeQuery = async () => {
+      let searchIntents: SearchIntent[] = [];
```

Comment:
This logic looks correct, however I find it a little difficult to follow since it requires me to go back and forth between the search-operations class to understand it. Would something like this work: `if ( state.location.userLocation ) {
   executeSearch()
} else {
   const intents = getSearchIntents()
   updateLocationIfNeeded(intents)
   executeSearch()
}`

Reply:
in both condition, it would still call execute search so I think it's better to just extracted out of that logic and call it once. But I updated the function to be updateLocationIfNeeded as suggested, hopefully that's easier to understand

Reply:
looks good!